AutomatedTesting/Gem/PythonTests/scripting/EditMenu_Default_UndoRedo.py

```diff
@@ -44,66 +52,68 @@ def EditMenu_Default_UndoRedo():
-    import ImportPathHelper as imports
-
-    imports.init()
-    from utils import Report
-
-    Report.start_test(EditMenu_Default_UndoRedo)
+    def __init__(self):
+        self.editor_main_window = None
+        self.sc_editor = None
+        self.sc_editor_main_window = None
+        self.variable_manager = None
+
+    def get_graph_variables_row_count(self, graph_variables):
+        return graph_variables.model().rowCount(QtCore.QModelIndex())
+
+
+    @pyside_utils.wrap_async
+    async def run_test(self):
+
+        # Preconditions
+        general.idle_enable(True)
+        scripting_tools.initialize_editor_object(self)
+
+        # 1) Open Script Canvas window
+        general.open_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
+
+        # 2) Get the SC window object
+        scripting_tools.initialize_sc_editor_objects(self)
+
+        # 3) Open Variable Manager if not opened already
+        scripting_tools.initialize_variable_manager_object(self)
+
+        # 4) Create Graph
+        scripting_tools.create_new_sc_graph(self.sc_editor_main_window)
+
+        # 5) Create new variable
+        add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, "addButton")
+        add_new_variable_button.click()  # Click on Create Variable button
+        # Select variable type
+        table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
+        model_index = pyside_utils.find_child_by_pattern(table_view, VARIABLE_TYPES[0])
+        # Click on it to create variable
+        pyside_utils.item_view_index_mouse_click(table_view, model_index)
+
+        # 6) Verify if the variable is created initially
+        graph_variables = self.variable_manager.findChild(QtWidgets.QTableView, GRAPH_VARIABLES_QT)
+        result = helper.wait_for_condition(
+            lambda: self.get_graph_variables_row_count(graph_variables) == 1, WAIT_TIME_3) # since we added 1 variable, rowcount=1
+        Report.result(Tests.variable_created, result)
+
+        # 7) Trigger Undo action and verify if variable is removed in Variable Manager
+        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Undo")
+        undo_redo_action.trigger()
+        result = helper.wait_for_condition(
+            lambda: self.get_graph_variables_row_count(graph_variables) == 0, WAIT_TIME_3) # since we triggered undo, rowcount=0
+        Report.result(Tests.undo_worked, result)
+
+        # 8) Trigger Redo action and verify if variable is readded in Variable Manager
+        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Redo")
+        undo_redo_action.trigger()
+        result = helper.wait_for_condition(
+            lambda: self.get_graph_variables_row_count(graph_variables) == 1, WAIT_TIME_3)  # since action is redone 1 variable is readded, rowcount=1
+        Report.result(Tests.redo_worked, result)
```

Comment:
Move the `helper.wait_for_condition(...)` into the `Report.result(...)`

Reply:
moved the helper function call into the report.result call